### PR TITLE
[iOS] Fix black box/playback issue when backgrounding fullscreen videos

### PR DESCRIPTION
**Pull request by kylehickinson**
Resolves https://github.com/brave/brave-browser/issues/36064

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/brave-ios/Sources/Brave/Frontend/Passcode/WindowProtection.swift [hunk 1]
```diff
@@ -86,18 +86,16 @@ public class WindowProtection {
...
   private var viewType: AuthViewType = .general
 
   private var isVisible: Bool = false {
     didSet {
       passcodeWindow.isHidden = !isVisible
+      // Prior window will be made key automatically after passcode window is dismissed
       if isVisible {
         passcodeWindow.makeKeyAndVisible()
-      } else {
-        protectedWindow.makeKeyAndVisible()
```
Comment (soner-yuksel) [inline ios/brave-ios/Sources/Brave/Frontend/Passcode/WindowProtection.swift:100]: First of all I tested the change both for youtube problem and when passcode is enabled etc

But the part I dont understand, dont we have to set active window with makeKeyAndVisible when passcode is dismissed ?

Comment (kylehickinson) [reply ios/brave-ios/Sources/Brave/Frontend/Passcode/WindowProtection.swift:100]: In my testing I found that It doesn't seem to be required, upon the `passcodeWindow` being hidden it simply gets removed from the window list and the last key window before showing it becomes key again

**Review by soner-yuksel — APPROVED** [review]
Great find.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `iPhone 11` running `iOS 17.3.1` using the following build(s):

```
Chromium | 1.65.63 Chromium: 122.0.6261.111 (Official Build) unknown (64-bit) 
--- | ---
Revision | fa36a45339ae...
OS | iOS
```

Using the STR/Cases outlined https://github.com/brave/brave-browser/issues/36064#issue-2134597210, reproduced the original issue using `1.62.1 (24.2.29.16)` as per the following:

IMG_0043

Using the same STR/Cases mentioned above, ensured that the issue wasn't reproducible using `1.65.63 Chromium: 122.0.6261.111`.

**Pull request merged**